native/src/main/java/io/ballerina/stdlib/http/api/client/actions/HttpClientAction.java

```diff
@@ -208,12 +208,7 @@ private static Object invokeClientMethod(Environment env, BObject client, BStrin
     private static Object invokeClientMethod(Environment env, BObject client, String methodName, Object[] paramFeed) {
         return env.yieldAndRun(() -> {
             try {
-                String strandParentFunctionName = Objects.isNull(env.getStrandMetadata()) ? null :
-                        env.getStrandMetadata().getParentFunctionName();
-                if (Objects.nonNull(strandParentFunctionName) && strandParentFunctionName.equals("onMessage")) {
-                    env.setStrandLocal(MAIN_STRAND, true);
```

Comment:
Dont we need to add this?

Reply:
I removed it because we didn't read it anywhere. Any specific reason we added this? @warunalakshitha

Reply:
It was there in old code too.

Reply:
+1 to remove